### PR TITLE
Bugfix/location responsibility optional

### PR DESCRIPTION
**Pull request by zeroplexer**
- Location responsibility optional
- Location and section order by name
- Allow to delete project description

### CODE REVIEW

File: src/main/java/io/github/bbortt/event/planner/web/rest/LocationResource.java [hunk 1]
```diff
@@ -127,9 +128,9 @@ public LocationResource(LocationService locationService, ProjectService projectS
 
     @GetMapping("/locations/project/{projectId}")
     @PreAuthorize("@projectService.hasAccessToProject(#projectId, \"" + RolesConstants.ADMIN + "\", \"" + RolesConstants.SECRETARY + "\")")
-    public ResponseEntity<List<Location>> getLocationsByProjectId(@PathVariable Long projectId) {
+    public ResponseEntity<List<Location>> getLocationsByProjectId(@PathVariable Long projectId, Sort sort) {
```
Comment (bbortt) [inline src/main/java/io/github/bbortt/event/planner/web/rest/LocationResource.java:131]: @zeroplexer you can actually just extend the existing API endpoint with a `Sort`. the default is `Sort.UNSORTED` which equals the database sort.

File: src/main/java/io/github/bbortt/event/planner/service/LocationService.java [hunk 2]
```diff
@@ -86,16 +89,16 @@ public void delete(Long id) {
      * @return the list of entities.
      */
     @Transactional(readOnly = true)
-    public List<Location> findAllByProjectId(Long projectId) {
+    public List<Location> findAllByProjectId(Long projectId, Sort sort) {
         log.debug("Request to get all Locations for Project {}", projectId);
-        return locationRepository.findAllByProjectId(projectId);
+        return locationRepository.findAllByProjectId(projectId, Sort.by(sort.stream().map(Order::ignoreCase).collect(Collectors.toList())));
```
Comment (bbortt) [inline src/main/java/io/github/bbortt/event/planner/service/LocationService.java:94]: case sensitivity was a problem. this little "hack" does the trick :)

**Review by bbortt — APPROVED** [review]

**Comment by bbortt** [comment]
closes #118.

File: src/main/java/io/github/bbortt/event/planner/web/rest/LocationResource.java (hunk 1 above)
Comment (zeroplexer) [reply src/main/java/io/github/bbortt/event/planner/web/rest/LocationResource.java:131]: Ah okey. Danke für den Hinweis.

**Pull request merged**